src/utils/config.ts

```diff
@@ -0,0 +1,464 @@
+}
+
+export async function removeFromGlobalConfig(targetName: string) {
+  let globalConfig = await getGlobalRcFile()
+  if (globalConfig && globalConfig.targets && globalConfig.targets.length) {
+    const targets = globalConfig.targets.filter(
+      (t: Target) => t.name !== targetName
+    )
+    await saveGlobalRcFile(JSON.stringify({ targets }, null, 2))
+  }
+}
+
+export async function getLocalConfig() {
+  const { buildSourceFolder } = getConstants()
+  let config = await getConfiguration(
+    path.join(buildSourceFolder, 'sasjsconfig.json')
+  )
+
+  return config
+}
+
+export async function saveToLocalConfig(target: Target) {
+  const { buildSourceFolder } = getConstants()
+  const targetJson = target.toJson()
+  let config = await getLocalConfig()
+  if (config) {
+    if (config.targets && config.targets.length) {
+      const existingTargetIndex = config.targets.findIndex(
+        (t) => t.name === target.name
+      )
+      if (existingTargetIndex > -1) {
+        config.targets[existingTargetIndex] = targetJson
+      } else {
+        config.targets.push(targetJson)
+      }
+    } else {
+      config.targets = [targetJson]
+    }
+  } else {
+    config = { targets: [targetJson] }
+  }
+
+  const configPath = path.join(buildSourceFolder, 'sasjsconfig.json')
+
+  await createFile(configPath, JSON.stringify(config, null, 2))
+
+  return configPath
+}
+
+export async function getFolders() {
+  const configPath = '../config.json'
+  const config = await readFile(path.join(__dirname, configPath))
+  if (config) {
+    const configJson = JSON.parse(config)
+    return Promise.resolve(configJson.folders)
+  }
+  return Promise.reject()
+}
+
+export async function getSourcePaths(buildSourceFolder: string) {
+  let configuration = await getConfiguration(
+    path.join(buildSourceFolder, 'sasjsconfig.json')
+  )
+  if (!configuration) {
+    configuration = { macroFolders: [] }
+  }
+
+  const sourcePaths = configuration.macroFolders
+    ? configuration.macroFolders.map((macroPath: string) =>
+        path.join(buildSourceFolder, macroPath)
+      )
+    : []
+  const macroCorePath = path.join(
+    process.projectDir,
+    'node_modules',
+    '@sasjs',
+    'core'
+  )
+  sourcePaths.push(macroCorePath)
+
+  return sourcePaths
+}
+
+/**
+ * Returns SAS program folders from configuration.
+ * This list includes both common and target-specific folders.
+ * @param {string} targetName - name of the configuration.
+ */
+export async function getProgramFolders(targetName: string) {
+  let programFolders: string[] = []
+  const projectRoot = await getProjectRoot()
+  const localConfig = await getConfiguration(
+    path.join(projectRoot, 'sasjs', 'sasjsconfig.json')
+  ).catch(() => null)
+  if (localConfig && localConfig.programFolders) {
+    programFolders = programFolders.concat(localConfig.programFolders)
+  }
+
+  const { target } = await findTargetInConfiguration(targetName)
+
+  if (!target) {
+    throw new Error(
```

Comment:
this piece of code is unreachable (and redundant as code throws in `findTargetInConfiguration` if there's no target.

Reply:
I've removed that bit of code.